packages/frontend/src/state/lp/hooks.ts

```diff
@@ -217,6 +217,164 @@ export const useCollectFees = () => {
+        tokenOut = weth
+      } else if (sqrtSqueethPrice.lt(sqrtLowerPrice)) {
+        // All squeeth position
+        // newLiquidity = positionEthValue/(squeethPrice/sqrt(lowerPrice) - squeethPrice/sqrt(upperPrice))
+        const liquidity = positionEthValue.div((squeethPrice.div(sqrtLowerPrice)).minus((squeethPrice.div(sqrtUpperPrice))))
+        wethAmountInLPAfter = new BigNumber(0)
+        // wPowerPerpAmount = L/sqrt(lowerPrice) - L/sqrt(upperPrice)
+        wPowerPerpAmountInLPAfter = (liquidity.div(sqrtLowerPrice).minus(liquidity.div(sqrtUpperPrice)))
+        amountIn = wethAmountInLPBefore
+        tokenIn = weth
+        tokenOut = oSqueeth
+      } else {
+        // newLiquidity = positionEthValue/(squeethPrice/sqrt(squeethPrice) - squeethPrice/sqrt(upperPrice) + sqrt(squeethPrice) - sqrt(lowerPrice))
+        const liquidity = positionEthValue.div((new BigNumber(squeethPrice).div(sqrtSqueethPrice))
+                                                                          .minus((new BigNumber(squeethPrice).div(sqrtUpperPrice)))
+                                                                          .plus(sqrtSqueethPrice)
+                                                                          .minus(sqrtLowerPrice))
+        // Calculate amounts of each asset to LP
+        // x = L(sqrt(upperPrice) - sqrt(squeethPrice))) / sqrt(squeethPrice) * sqrt(upperPrice)
+        // y = L(sqrt(squeethPrice) - sqrt(lowerPrice))
+        wPowerPerpAmountInLPAfter = liquidity.times(sqrtUpperPrice.minus(sqrtSqueethPrice)).div((sqrtSqueethPrice.times(sqrtUpperPrice)))
+        wethAmountInLPAfter = liquidity.times(sqrtSqueethPrice.minus(sqrtLowerPrice))
+        const needMoreWeth = new BigNumber(wethAmountInLPBefore).lt(new BigNumber(wethAmountInLPAfter))
+        const needMoreSqueeth = new BigNumber(wPowerPerpAmountInLPBefore).lt(new BigNumber(wPowerPerpAmountInLPAfter))
+        tokenIn = needMoreWeth ? oSqueeth : weth
+        tokenOut = needMoreWeth ? weth : oSqueeth
+        amountIn = needMoreWeth && !needMoreSqueeth ? new BigNumber(wPowerPerpAmountInLPBefore).minus(new BigNumber(wPowerPerpAmountInLPAfter)).toFixed(0) :
+                   needMoreSqueeth && !needMoreWeth ? new BigNumber(wethAmountInLPBefore).minus(new BigNumber(wethAmountInLPAfter)).toFixed(0)
+                  : 0
+      }
+
+      // Get amount mins
+      const amount0New = isWethToken0 ? wethAmountInLPAfter : wPowerPerpAmountInLPAfter
+      const amount1New = isWethToken0 ? wPowerPerpAmountInLPAfter : wethAmountInLPAfter
+      const amount0MinNew = amount0New.times(new BigNumber(1).minus(slippage)).toFixed(0)
+      const amount1MinNew = amount1New.times(new BigNumber(1).minus(slippage)).toFixed(0)
+
+      // Get limit price
+      const amountOut = await getExactIn(new BigNumber(amountIn), tokenIn == oSqueeth)
+      const limitPrice = new BigNumber(amountOut).div(new BigNumber(amountIn)).times(new BigNumber(1).minus(slippage))
```

Comment:
This needs splitting into two case

I think limit price needs to be split out for the two cases.

needMoreWeth = true:  selling oSQTH for ETH, amountIn is oSQTH, amountOut is ETH  (accept slightly lower price to sell oSQTH)  
const limitPrice = (amountOut / amountIn) * (1 - slippage)
needMoreSqueeth = true: selling ETH for oSQTH, amountIn is WETH, amountOut is oSQTH (accept slightly higher price to buy oSQTH)
const limitPrice = (amountIn / amountOut) * (1 + slippage)